### PR TITLE
adding a decider onto middleware so that logs can be surpessed

### PR DESCRIPTION
**Pull request by domgreen**
Adding a `decider` method so that we can decide that we do not want to log messages for certain calls.

### CODE REVIEW

**Comment by domgreen** [comment]
@yifanzz 👍

File: logging/logrus/middleware_test.go [hunk 1]
```diff
@@ -33,6 +33,12 @@ func TestLogrusMiddlewareSuite(t *testing.T) {
 	s.WaresTestSuite.ServerMiddleware = []httpwares.Middleware{
 		http_logrus.Middleware(
 			logrus.NewEntry(s.logrusBaseTestSuite.logger).WithField("http.handler.group", "my_service"),
+			http_logrus.WithDecider(func(w httpwares.WrappedResponseWriter, r *http.Request) bool {
+				if r.URL.Path == "/blah" {
+					return false
+				}
+				return true
```
Comment (yifanzz) [inline logging/logrus/middleware_test.go]: cough cough

File: logging/logrus/options.go [hunk 2]
```diff
@@ -95,6 +98,16 @@ func WithResponseBodyCapture(deciderFunc func(r *http.Request, status int) bool)
 	}
 }
 
+// WithDecider customizes the function for deciding if the middleware logs at the end of the request.
+func WithDecider(f Decider) Option {
+	return func(o *options) {
+		o.shouldLog = f
```
Comment (yifanzz) [inline logging/logrus/options.go]: how about if we went for something more specific and have a option for `PathToExclude`? I don't think we need a function this generic atm.

Comment (domgreen) [reply logging/logrus/options.go]: We have a specific use case where we want to suppress logs where the status code is between `400` and `500`.

File: logging/logrus/middleware.go [hunk 3]
```diff
@@ -35,8 +35,12 @@ func Middleware(entry *logrus.Entry, opts ...Option) httpwares.Middleware {
 			})
 			startTime := time.Now()
 			nextHandler.ServeHTTP(wrappedResp, newReq)
-			capture.finish() // captureResponse has a nil check, this can be nil
 
+			if !o.shouldLog(wrappedResp, newReq) {
```
Comment (yifanzz) [inline logging/logrus/middleware.go]: if you do `if o.shouldLog != nil && !o.shouldLog(....) {...}` then you won't need to have the default decider.

**Pull request merged**